kaldb/src/main/java/com/slack/kaldb/server/RecoveryTaskCreator.java

```diff
@@ -184,7 +188,40 @@ public long determineStartingOffset(long currentHeadOffsetForPartition) {
 
     if (highestDurableOffsetForPartition <= 0) {
       LOG.info("There is no prior offset for this partition {}.", partitionId);
-      return highestDurableOffsetForPartition;
+
+      // If the user wants to start at the current offset in Kafka and _does not_ want to create
```

Comment:
You might wanna reformat this comment block, as the auto-formatter seems to have wrecked it a bit.